virttest/utils_time.py

```diff
@@ -0,0 +1,229 @@
+    except process.CmdError:
+        logging.warn("Can not get host's timezone")
+    tz = re.match(r'^(?:\s+Time zone:\s)(\w+\/\w+|UTC)(?:\s\(\w+,\s)([+|-]'
+                  r'\d{4})\)$', o, re.M | re.I)
+    if tz:
+        timez_city = tz.groups()[0]
+        timez_code = tz.groups()[1]
+    else:
+        timez_city = ""
+        timez_code = ""
+    return (timez_city, timez_code)
+
+
+@error_context.context_aware
+def align_guestz_to_hostz(params, vm, session, zonecode, zonecity):
+    """
+    Provide timezone alignment between host and guest.
+
+    This function get the guest's timezone and compare it with host's
+    timezone, if they are same, return True, or change the guest's
+    timezone to the timezone same with host's utc code.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: Virtual machine for vm.
+    :param session: VM session
+    :param zonecode: Host's timezone in code mode.
+    :param zonecity: Host's timezone in city mode.
+
+    :return: True if guest has same timezone with host.
+    """
+    status = -1
+    o_guest = session.cmd_output(params["check_tz_cmd"])
+    logging.info("check timezone in guest: %s" % o_guest)
+    if params.get("os_type") == "linux":
+        guest_tz = re.match(r'^(?:\s+Time zone:\s)(\w+\/\w+|UTC)(?:\s\(\w+'
+                            r',\s)([+|-]\d{4})\)$', o_guest, re.M | re.I)
+        if guest_tz.groups()[1] == zonecode:
+            return (True, session)
+        else:
+            status = session.cmd_status(params["set_tz_cmd"] % zonecity)
+    elif params.get("os_type") == "windows":
+        guest_tzs = []
+        guest_tz = []
+        tz_list = session.cmd_output(params["list_tz_cmd"])
+        tz_list = tz_list.splitlines()
+        for line in tz_list:
+            if not line:
+                continue
+            og = re.search(r'(?:\(UTC([+|-]\d{2}:\d{2})?)', line,
+                           re.M | re.I)
+            if og and og.group(1):
+                og_ = re.sub(r'(\d{2}):(\d{2})', r'\1\2', og.group(1))
+                guest_tz.append(og_)
+                continue
+            elif og:
+                guest_tz.append("+0000")
+                continue
+            og_value = re.search(r'(\S+(?:\s\S+)*)', line, re.M | re.I)
+            if og_value:
+                guest_tz.append(og_value.group(0))
+            guest_tzs.append(guest_tz)
+            guest_tz = []
+
+        for value in guest_tzs:
+            if value[0] == zonecode:
+                return (True, session)
+
+        # Windows guest's timezone not match host's timezone
+        # get the timezone value for windows guest
+        tz_value = ""
+        for value in guest_tzs:
+            if zonecode == value[0]:
+                tz_value = value[1]
+                break
+        status = session.cmd_status(params["set_tz_cmd"] % tz_value)
+
+    if not status:
+        session = vm.reboot(session=session, timeout=600)
```

Comment:
If set time zone failed, reboot the guest and return True directly?

Reply:
Thanks, update it.